### DISCUSSION TITLE
Modifying presenter view layout

### DISCUSSION THREAD

**Discussion by cfcurtis** [marp-team/marp#261] [Q&A] [answered]
Hi! I'm really enjoying using Marp for my class this semester, it makes it very easy to put together a good-looking presentation and also render everything as a single page on my course website.

I would like to tweak the presenter view so that the "coming up next" slide takes up a greater proportion or all of the view. I use presenter view in half my screen, Google Meet in the other half, and the actual presentation is on my second screen. As a result, I see 3 copies of the current slide and only one hard-to-read version of the next slide.

I was poking through the marp/marp-cli repo and it looks like [this](https://github.com/marp-team/marp-cli/blob/1ef5ffd5a27f4bc574ebb2af9702c98276c2db84/src/templates/bespoke/bespoke.scss#L197) is where I should begin, but I'm not sure how to override the Bespoke css that gets embedded in the slide.

**Comment by yhatt** [comment]
Thank you for using Marp! But unfortunately our presenter view is minimal and not customizable via overloading styles, because HTML output in Marp official tools is just a bonus. (Our goal is making Markdown presentation ecosystem for making reproducible slides in PDF and providing a library that easy to integrate with the own slide experience by developers)

You have to make your own Marp CLI build from the source if you like to customize `bespoke.scss` in Marp CLI.

If you think that's too much, using your own bookmarklet (can make at https://www.squarefree.com/userstyles/make-bookmarklet.html) or the user style extension for the browser to modify the presenter view layout is much better.

1. Make your own bookmarklet with including custom style like this at [here](https://www.squarefree.com/userstyles/make-bookmarklet.html):

   ```css
   .bespoke-marp-presenter-container {
     grid-template:
       'current next' 4fr
       'current note' 1fr
       'info    note' 3em /
       0 minmax(0, 1fr) !important;
   }
   ```

2. Open a presentation in 2 tabs with query string, `presentation.html?sync=uniqueKeyAsYouLike` and `presentation.html?sync=uniqueKeyAsYouLike&view=presenter`. `sync` query parameter is required to sync the page between tabs that a unique key have matched.
3. Apply created bookmarklet to the presenter view.

**Reply by cfcurtis** [reply]
Thank you, this works! I may look in to the custom CLI thing if I find a significantly improved experience, but I'm running marp through a github action so it makes it a little bit more complicated.

**Comment by yhatt** [comment, marked as answer]
**UPDATE**: [Marp CLI v1.7.0](https://github.com/marp-team/marp-cli/releases/tag/v1.7.0) is now making draggable the vertical splitter of presenter view, with awesome contributions by @chrisns. You can maximize notes just by moving splitter to left :smile:

![](https://user-images.githubusercontent.com/3993388/155343724-4d2f3aee-c1c6-4ef8-8a6d-0dd4e06bce46.gif)
